appsec/src/helper/network/acceptor.cpp

```diff
@@ -80,13 +80,10 @@ socket::ptr acceptor::accept()
     struct sockaddr_un addr {};
     socklen_t len = sizeof(addr);
 
-    int s;
-    do {
-        // NOLINTNEXTLINE
-        s = ::accept(sock_, reinterpret_cast<struct sockaddr *>(&addr), &len);
-    } while (s == -1 && errno == EINTR);
+    // NOLINTNEXTLINE
+    int s = ::accept(sock_, reinterpret_cast<struct sockaddr *>(&addr), &len);
     if (s == -1) {
-        if (errno == EAGAIN) {
+        if (errno == EINTR || errno == EAGAIN) {
```

Comment:
nit: EINTR in this case isn't really caused by a timeout is it? So perhaps a different exception would make sense, although in a practical sense it doesn't matter.

Reply:
actually I really don't like the exception here. I think the only reason it uses an exception is because an empty result is taken to have a different meaning. But I'll change it for a more complex return value